### DISCUSSION TITLE
class members

### DISCUSSION THREAD

**Discussion by gasagna** [jgromes/RadioLib#698] [General]
Hi, 

I am developing a library that sits on top of RadioLib and uses the available radio drivers to allow addressed, acknowledged packetised communication between LoRa devices, a-la RadioHead. 

One of the classes I have looks something like this
```c++
class RadioLibManager {
    public:
        RadioLibManager(PhysicalLayer& radio)
    private:
        PhysicalLayer& _radio;
}
```
where I have one class member for the radio driver.

However, `PhysicalLayer` does not define all functions I need, and seems to be missing some part of the interface. For instance, the functions
```c++
sleep
getRSSI
getSNR
clearDio1Action
setDio1Action
startReceive
```
are defined in some of the subclasses, e.g. they are defined for the `sx1262` radios I am currently testing.

Maybe I am missing something, but why does `PhysicalLayer` not define the full interface? 

BTW, thanks a lot for the work you have put in this library and the continuous support and updates you provide.

**Comment by jgromes** [comment]
`PhysicalLayer` is basically just an abstraction layer to allow usage of protocols such as RTTY, SSTV etc. independently of the specific driver (SX126x, SX127x) etc. It also contains some generic common functionality (e.g. converting to/from Arduino String class). But you're right, it should have the complete API. There's not particular reason it's not complete, there just wasn't a use case for it.

**Reply by gasagna** [reply]
I am happy to produce a PR that would define the functions I need.

**Reply by jgromes** [reply]
That would be very much appreciated! Please follow the interface that was already done for e.g. `PhysicalLayer::setDIOMapping` - make the methods virtual and provide a default implementation which returns `RADIOLIB_ERR_UNSUPPORTED`.

**Comment by gasagna** [comment]
Linking this discussion to the PR #700!
